packages/runner/lib/usage.ts

```diff
@@ -0,0 +1,57 @@
+import os from 'os';
+import { createActivityLogMessage } from '@nangohq/shared';
+
+interface Tracked {
+    activityLogId: number;
+    environmentId: number;
+}
+
+const MEMORY_WARNING_PERCENTAGE_THRESHOLD = 70;
+let lastMemoryReportDate: Date | null = null;
+const tracked: Map<number, Tracked> = new Map<number, Tracked>();
```

Comment:
While I believe our usage of Class is not always clear, I think it was a good time to avoid floating state imo